Language/Classes/Instance_Methods/Method_noSuchMethod/forwarder_A01_t07.dart

```diff
@@ -0,0 +1,150 @@
+/// and accessible to L that correctly overrides S (that is, no member named m
+/// is declared or inherited by C, or one is inherited, but it does not have the
+/// required signature). In this case we also say that S is noSuchMethod
+/// forwarded.
+///
+/// • Forced by privacy: There exists a direct or indirect superinterface D of C
+/// which is declared in a library L2 different from L, the interface of D
+/// contains a member signature S named m, m is a private name, and no
+/// superclass of C has a concrete member named m accessible to L2 that
+/// correctly overrides S. In this case we also say that S is noSuchMethod
+/// forwarded.
+///
+/// For a concrete class C, a noSuchMethod forwarder is implicitly induced for
+/// each member signature which is noSuchMethod forwarded.
+///
+/// @description Checks that for a concrete class, a `noSuchMethod` forwarder is
+/// implicitly induced for each member signature which is `noSuchMethod`
+/// forwarded. Test the case when forwarding is requested in program and `C` has
+/// no concrete member named `m`. Test setters
+/// @author sgrekhov22@gmail.com
+
+import "../../../../Utils/expect.dart";
+
+String log = "";
+
+abstract mixin class A {
+  void set a(String s);
+}
+
+class C1 {
+  void set m(String s);
+  noSuchMethod(Invocation i) {
+    log = "C1";
+  }
+}
+
+class C2 extends A {
+  dynamic noSuchMethod(Invocation i) {
+    log = "C2";
+  }
+}
+
+class C3 implements A {
+  dynamic noSuchMethod(Invocation i) {
+    log = "C3";
+  }
+}
+
+class C4 with A {
+  dynamic noSuchMethod(Invocation i) {
+    log = "C4";
+  }
+}
+
+mixin M1 {
+  void set m(String s);
+  noSuchMethod(Invocation i) {
+    log = "M1";
+  }
+}
+
+mixin M2 on A {
+  dynamic noSuchMethod(Invocation i) {
+    log = "M2";
+  }
+}
+
+mixin M3 implements A {
+  dynamic noSuchMethod(Invocation i) {
+    log = "M3";
+  }
+}
+
+class MA1 = Object with M1;
+class MA2 = A with M2;
+class MA3 = A with M3;
+
+enum E1 {
+  e1,
+  e2;
+
+  void set m(String s);
+  noSuchMethod(Invocation i) {
+    log = "E1";
+  }
+}
+
+enum E2 implements A {
+  e1,
+  e2;
+
+  dynamic noSuchMethod(Invocation i) {
+    log = "E2";
+  }
+}
+
+enum E3 with A {
+  e1,
+  e2;
+
+  dynamic noSuchMethod(Invocation i) {
+    log = "E3";
+  }
+}
+
+main() {
+  C1().m = "";
+  Expect.equals("C1", log);
+  log = "";
+  C2().a = "";
+  Expect.equals("C2", log);
+  log = "";
+  C3().a = "";
+  Expect.equals("C3", log);
+  log = "";
+  C4().a = "";
+  Expect.equals("C4", log);
+  log = "";
+  MA1().m = "";
+  Expect.equals("M1", log);
+  log = "";
+  MA2().a = "";
+  Expect.equals("M2", log);
+  log = "";
+  MA2().a = "";
+  Expect.equals("M2", log);
+  log = "";
+  MA3().a = "";
+  Expect.equals("M3", log);
+  log = "";
+  E1.e1.m = "";
+  Expect.equals("E1", log);
+  log = "";
+  E2.e1.a = "";
+  Expect.equals("E2", log);
+  log = "";
+  E3.e1.a = "";
+  Expect.equals("E3", log);
+  log = "";
```

Comment:
I think it should be possible to make the distinction between the case where we actually have a noSuchMethod forwarder for the setter, and the case where `noSuchMethod` is invoked directly, if we could pass an argument to the setter which is type incorrect (but statically typed as `dynamic`): `noSuchMethod` would be happy to receive that argument, but an actual forwarder should throw.

Reply:
Added `forwarder_A01_t09.dart` testing it

Reply:
Thanks!